optaplanner-operator/src/main/java/org/optaplanner/operator/impl/solver/model/Solver.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.operator.impl.solver.model;
+
+import org.optaplanner.operator.impl.solver.model.messaging.MessageAddress;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import io.fabric8.kubernetes.api.model.Namespaced;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.model.annotation.Group;
+import io.fabric8.kubernetes.model.annotation.Version;
+
+@Group("org.optaplanner.solver")
+@Version("v1")
+public final class Solver extends CustomResource<SolverSpec, SolverStatus> implements Namespaced {
```

Comment:
Let's not name a class Solver, it's confusing with the Solver class in optaplanner-core.

Reply:
@ge0ffrey Right; let's have a look at what alternatives we have:
A) OptaPlannerSolver, OptaPlannerSolverStatus
B) SolverResource, SolverStatusResource

If we again take a look at Strimzi, they have:
KafkaTopic, KafkaTopicStatus (so it's the option A)

Reply:
In the end, I decided on A), but just for the classes. The name of the k8s custom resource remains the same: the `solver`. Let me know if we need to discuss the naming more thoroughly.